### ISSUE TITLE
Docs: Incorrect uses of `@retval`

### ISSUE THREAD

**Issue by Thalley** [zephyrproject-rtos/zephyr#65974]
**Describe the bug**
The `@retval` keyword uses the first word as the value, and should only be used like `@retval 0 <text>` or `@retval -EINVAL <text>`. There are several functions that are using it incorrectly, where the first word is describing the return value instead. 

For example https://docs.zephyrproject.org/latest/kernel/services/synchronization/events.html#c.k_event_test where the return values part now looks weird: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/2520413/45e28172-f7bf-422b-9c2b-fe1831063e76)

There are also cases where a function may return a set of specific error codes if negative, but an actual value if positive like https://docs.zephyrproject.org/latest/connectivity/bluetooth/api/audio.html#c.bt_audio_codec_cfg_set_frame_blocks_per_sdu. I am unsure what the correct solution for those functions are, but maybe they should similar to https://docs.zephyrproject.org/latest/hardware/cache/index.html#c.sys_cache_instr_line_size_get? 

Lastly, there are also cases where `-errno` is provided as the `@retval` which isn't really a value either. 

**To Reproduce**
N/A

**Expected behavior**
Expect the `@retval`s to be more specific in terms of the value that is returned, and specific in a way where the first word is correctly used as the value. 

**Impact**
Minor.

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: ea887af0aa112b1e707be84a299788a789b0c129

**Additional context**
N/A

**Comment by kartben** [comment]
Thanks for opening this!

> [...] I am unsure what the correct solution for those functions.

I *think* it should be both `@return` and `@retval`, like https://docs.zephyrproject.org/latest/hardware/peripherals/uart.html#c.uart_fifo_fill

I am going to check how other projects do it and/or if there are guidelines coming from Doxygen regarding how to address this.

**Comment by kartben** [comment]
@Thalley pretty sure I will end up using https://docs.rtems.org/docs/main/eng/coding-doxygen.html#function-declarations as a starting point when I will finally find time to add some doxygen guidelines to our docs

**Comment by qbx11** [comment]
Hi, I'd like to contribute a fix for this issue and have a question.
                                                                                                                                                                                                               
I  see two patterns being used  for negative error returns:
                                                                                                                                                                                                               
  - `@retval <0 Other negative errno on failure.`
  - `@retval -errno Other negative errno on failure.`                                                                                                                                 
                                                                           
 The issue description mentions that `-errno` is  kinda incorrect. However, #98324 left existing `-errno` entries unchanged. Should `-errno`  be replaced with `<0` or is it acceptable?

**Comment by Thalley** [comment]
> The issue description mentions that `-errno` is kinda incorrect. However, [#98324](https://github.com/zephyrproject-rtos/zephyr/pull/98324) left existing `-errno` entries unchanged. Should `-errno` be replaced with `<0` or is it acceptable?

I'd say neither are acceptable; IMO `@retval` should be used with a specific value, e.g. `5`, `0`, `-EINVAL`, etc., and not expressions like `< 0` or references to groups of values like `errno`. The purpose of using `retval` is to document precisely which values the caller can expect

**Comment by kartben** [comment]
> > The issue description mentions that `-errno` is kinda incorrect. However, [#98324](https://github.com/zephyrproject-rtos/zephyr/pull/98324) left existing `-errno` entries unchanged. Should `-errno` be replaced with `<0` or is it acceptable?
> 
> I'd say neither are acceptable; IMO `@retval` should be used with a specific value, e.g. `5`, `0`, `-EINVAL`, etc., and not expressions like `< 0` or references to groups of values like `errno`. The purpose of using `retval` is to document precisely which values the caller can expect

Agreed. We should move towards having one (usually the only) `@return` statement that ends with "..., or a negative error code on error.", and make sure that the known/important negative error codes are actually documented when it makes sense.

**Comment by qbx11** [comment]
I've opened a PR (#107276) to start fixing the @retval misuses, beginning with `include/zephyr/pm/`.  I wanted to validate the approach before I continue. Does the approach above look correct to you @kartben, @Thalley?
